api/users/usersController.js

```diff
@@ -42,5 +43,46 @@ module.exports.factory = (
       })
   })
 
+  router.put('/users/:email/addcat', function (req, res) {
+    const body = req.body
+
+    Promise.resolve(body)
+      .then(() => new Promise(getUser(req.params.email)))
+      .then(user => {
+        if (!user) {
+          throw new Error('A user with that email address does not exist')
+        }
+
+        return body
+      })
+      .then(body => new Promise(addCategory(req.params.email, body)))
+      .then(response => {
+        res.status(200).send(response)
+      }).catch(err => {
+        logger.error(err)
+        res.status(400).send({ messages: [err.message] })
+      })
+  })
+
+  router.put('/users/:email/remcat', function (req, res) {
```

Comment:
Similar suggestion. Depending on how you want to implement it, you can restructure and make this a DELETE instead of a PUT. https://stackoverflow.com/questions/37138720/restful-api-delete-some-of-a-resource-collection This link shows how you could specify a DELETE and delete one category such as DELETE '/users/:email/category?category=fiction' This would say delete fiction from the user's list of categories

Reply:
With this delete you will no longer need a body. You will instead be getting the category from the request params.